enterprise/reporting/flusher/db/db.go

```diff
@@ -0,0 +1,17 @@
+//go:generate mockgen -destination=./db_mock.go -package=db -source=./db.go DB
+package db
+
+import (
+	"context"
+	"time"
+
+	"github.com/rudderlabs/rudder-server/enterprise/reporting/flusher/report"
+)
+
+type DB interface {
+	InitDB() error
+	GetStart(ctx context.Context, table string) (time.Time, error)
+	FetchBatch(ctx context.Context, table string, start, end time.Time, limit, offset int) ([]report.RawReport, error)
+	Delete(ctx context.Context, table string, start, end time.Time) error
+	CloseDB() error
+}
```

Comment:
I would move the interface to flusher instead. 

Also we can avoid InitDB, also consider if CloseDB should be part of the interface or not.

```suggestion
type DB interface {
	GetStart(ctx context.Context, table string) (time.Time, error)
	FetchBatch(ctx context.Context, table string, start, end time.Time, limit, offset int) ([]report.RawReport, error)
	Delete(ctx context.Context, table string, start, end time.Time) error
}
```